docs/README.md

```diff
@@ -0,0 +1,325 @@
+We are, indeed, leveraging the good work that CoreOS folks have done so much so
+that we're using their patches for the GRUB bootloader to provide rootfs upgrade
+functionality between IMGA and IMGB root filesystems. Unlike CoreOS, though, EVE
+only relies on the partition symbolic names (and not offsets in the GPT) and hence
+can extend any existing GPT that it can recognize. This comes handy on some of
+the ARM platforms that reserve a few of the GPT entries to host things like
+firmware. Finally, it must be said that EVE doesn't support legacy partitioning
+schemes (such as MBR).
+
+EVE's root filesystem is hosted in both IMGA and IMGB partitions. This allows
+a running EVE instance to safely update itself. If you want to know more about
+how this process works read [baseimage update](BASEIMAGE-UPDATE.md) documentation.
+
+Both IMGA and IMGB host a filesystem containing the following software components
+
+* Second stage [GRUB bootloader](../pkg/grub) and [its configuration](../pkg/grub/rootfs.cfg) in /EFI folder
+* [Hypervisor](../pkg/xen) and [Linux Kernel](../pkg/kernel) in /boot folder
+* [EVE microservices](../pkg/pillar) in /containers folder
+* Minimalistic [Alpine Linux](https://alpinelinux.org)-derived UNIX environment
+
+EVE microservices are ultimately responsible for providing the entire EVE
+experience and they are described in greater details below.
+
+## Runtime Lifecycle
+
+EVE boots as any traditional operating system and it requires UEFI firmware.
+EVE's boot process starts with UEFI firmware executing the GRUB trampoline
+that resides in GPT entry called ```EFI System```. This instance of GRUB is
+the only component that can *not* be changed after the initial installation
+of EVE on an Edge Node and as a result it is kept extremely simple. The only
+role that a GRUB trampoline [plays](../pkg/mkimage-raw-efi/efi-files/EFI/BOOT/grub.cfg)
+is selecting whether to chainload second stage GRUB from IMGA or IMGB partitions.
+
+Second stage GRUB is expected to do all the [heavy lifiting](../pkg/grub/rootfs.cfg)
+of actually booting an EVE instance and because it resides in IMGA or IMGB
+partitions it can easily be upgraded and patched.
+
+After second stage GRUB is done loading type-1 hypervisor and Control Domain
+kernel the rest of the runtime sequence relies solely on what happens with EVE
+microservices. All of these services are available under */containers/services*
+folder in IMGA or IMGB root filesystem. Their lifecycle is currently managed
+by [linuxkit init system](https://github.com/linuxkit/linuxkit/tree/master/pkg/init)
+and monitored by a built-in [software watchdog](../pkg/watchdog). Software
+watchdog, in turn, [relies](../pkg/pillar/scripts/device-steps.sh#L56) on
+Edge Node's hardware watchdog to cope with unexpected software failures.
+
+All EVE microservices are expected to communicate with each other by using
+our own [IPC mechanism](COMMS.md).
```

Comment:
"by using the EVE dedicated IPC mechanism"

Reply:
fixed